crates/polars-core/src/chunked_array/cast.rs

```diff
@@ -174,12 +174,22 @@ impl ChunkCast for Utf8Chunked {
     fn cast(&self, data_type: &DataType) -> PolarsResult<Series> {
         match data_type {
             #[cfg(feature = "dtype-categorical")]
-            DataType::Categorical(_) => {
-                let iter = self.into_iter();
-                let mut builder = CategoricalChunkedBuilder::new(self.name(), self.len());
-                builder.drain_iter(iter);
-                let ca = builder.finish();
-                Ok(ca.into_series())
+            DataType::Categorical(rev_map) => match rev_map {
+                None => {
+                    let iter = self.into_iter();
+                    let mut builder = CategoricalChunkedBuilder::new(self.name(), self.len());
+                    builder.drain_iter(iter);
+                    let ca = builder.finish();
+                    Ok(ca.into_series())
+                },
+                Some(rev_map) => {
+                    polars_ensure!(rev_map.is_enum(), InvalidOperation: "casting to a non-enum variant with rev map is not supported for the user");
+                    CategoricalChunked::from_utf_to_enum(self, rev_map.get_categories()).map(|ca| {
```

Comment:
I think we should name this `utf8_to_enum` or `str_to_enum` as we don't support generic `utf` variants.